graphql/util.go

```diff
@@ -378,31 +378,18 @@ func modifyVersionHandler(ctx context.Context, patchID string, modification mode
 	}
 
 	if evergreen.IsPatchRequester(v.Requester) {
-		// restart is handled through graphql because we need the user to specify
-		// which downstream tasks they want to restart
+		// Restart is handled through graphql because we need the user to specify
+		// which downstream tasks they want to restart.
 		if modification.Action != evergreen.RestartAction {
-			//do the same for child patches
-			p, err := patch.FindOneId(patchID)
+			// Only modify the child patch if it is finalized.
+			childPatches, err := patch.GetFinalizedChildPatchIdsForPatch(patchID)
```

Comment:
When testing restarts in staging, can you please include scenarios with child patches?

Reply:
This actually doesn't affect restarts since we specifically only do this if it's not a restart. I can add a unit test to verify this new function though.

Reply:
I'd prefer to rely on unit testing in this case, since the change that was made was small and query related so it can be verified in a repeatable way. I'll push this shortly!